### DISCUSSION TITLE
Use on non-FreeRTOS builds

### DISCUSSION THREAD

**Discussion by rayz55** [eModbus/eModbus#38] [General]
I set out to find a good modbus implementation for use in Arduino projects. When I couldn't find any that were both well designed and copy-right, I started to build my own. But then I ran across eModbus and really like the way it's designed. But it assumes FreeRTOS, which most of my projects don't use.

It looks like I could replace the threads with a polling interface with medium complexity. But maybe there's a better way? Any suggestions from the group?

Thanks!

**Comment by Miq1** [comment]
You are very welcome to add a non-task version to the library. Both of us mainly are working with the ESP32, so the need did not rise yet. I am, though, currently implementing a smart plug firmware that is supposed to offer energy meter capabilities via Modbus. As the MCU is an ESP8285 in this case, I may need a single-task implementation as well.

The TCP version will be easier to implement first, as the RS485 bus timing can be critical if not maintained in a dedicated task.

**Reply by bertmelis** [reply]
The async version is suitable for ESP8266 with minor changes. Which mcu are you aiming at?

**Reply by rayz55** [reply]
The initial version will be on a ATSAME53N19A (ARM Cortex-M4F @120MHz).
